### ISSUE TITLE
__repr__ not defined for classes

### ISSUE THREAD

**Issue by chlowell** [Azure/azure-sdk-for-python#7232]
For example, from the `IndexManagement` sample:
```
>>> created_db = CreateDatabaseIfNotExists(client, DATABASE_ID)
>>> print(created_db)
<azure.cosmos.database.DatabaseProxy object at 0x000002A48509EB38>

>>> created_Container = db.create_container(id=CONTAINER_ID, partition_key=PARTITION_KEY)
>>> print(created_Container)
<azure.cosmos.container.ContainerProxy object at 0x000002A4851F1D68>
```

**Comment by bryevdv** [comment]
@chlowell I did not find much precent for reprs in storage, keyvault, or eventhubs, except in models modules. I have a branch ready (waiting on another PR to clear up some latent linter issues on master) that does this:
```
In [10]: c
Out[10]: <CosmosClient [https://brvandev-cosmos.documents.azure.com:443/]>

In [11]: c.get_database_client("foo")
Out[11]: <DatabaseProxy [dbs/foo]>

In [12]: c.get_database_client("foo").get_container_client("bar")
Out[12]: <ContainerProxy [dbs/foo/colls/bar]>

In [13]: PartitionKey("stuff")
Out[13]: <PartitionKey [stuff]>
```
Is that what you had in mind?

**Comment by chlowell** [comment]
Yes, that's the idea. My heuristic for Key Vault was, make the representation an `eval`able recreation of the object if feasible else `<ClassName [identifier or something important]>`. As I recall the design guidelines only stipulate `__repr__` be defined and never return more than 1024 characters.

**Comment by bryevdv** [comment]
@chlowell Thanks, with the exception of `CosmosClient`, all the other classes are not intended to be directly instantiated by users (and `CosmosClient` may have long, complicated args including credentials) so I will go with the style above.

**Comment by bryevdv** [comment]
closed by https://github.com/Azure/azure-sdk-for-python/pull/8241
